### DISCUSSION TITLE
How to disabled button , When clicking on a button in Actions from the listing.

### DISCUSSION THREAD

**Discussion by jenishprajapati-kombee** [Power-Components/livewire-powergrid#1276] [Q&A]
How to disabled button , When clicking on a button in Actions from the listing.

![image](https://github.com/Power-Components/livewire-powergrid/assets/67471303/7ea30d6d-5f46-4cc6-93f9-514ea582fbb1)

**Comment by abhaypithadiya** [comment]
Hey! 

Check out the documentation here: https://livewire-powergrid.com/table/action-rules.html#disable

Hope this helps!

**Comment by jenishprajapati-kombee** [comment]
Hello abhaypithadiya,
Thanks for the reply. Yes I have read the documentation but want to achieve like button should be clikable only once, I am facing issues curretly multiple clicks are allowed.

**Comment by abhaypithadiya** [comment]
Hey!

One way you can achieve this is:

1. Create an array to save the ids which are clicked.
```
public $addedProducts = [];
```

2. Create a function in your actions which will save the ids, if you already have a function you can customise the function.
```
    public function actions(Product $row): array
    {
        return [
            Button::add(button_name)
                ->render(function (Model $row) {
                        return Blade::render(
                            <<<HTML
                        <button wire:click.prevent='addProduct($row)'></button>
                        HTML
                        );
                }),
        ];
    }
```

3. Create a the addProduct function.
```
    public function addProduct($data)
    {
        $checkIfProductIsSelected = in_array($data['id'], $this->addedProducts);

        if (! $checkIfProductIsSelected) {
            $this->addedProducts[] = $data['id'];
        }
    }
```

4. Lastly, use the rules to disable the ids in the array
```
    public function actionRules(): array
    {
        return [
            Rule::checkbox()
                ->when(fn ($product) => in_array($product->id, $this->addedProducts))
                ->disable(),
        ];
    }
```

Play around the code as it fits with your liking. Let me know if this works.
